lib/node_modules/@stdlib/blas/ext/base/dsortnans/lib/dsortnans.js

```diff
@@ -0,0 +1,57 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var stride2offset = require( '@stdlib/strided/base/stride2offset' );
+var ndarray = require( './ndarray.js' );
+
+
+// MAIN //
+
+/**
+* Partitions a double-precision floating-point strided array by moving all NaNs either to the beginning or the end of the array.
+*
+* ## Notes
+*
+* -   The algorithm is not stable, meaning that the relative order of non-NaN elements is not guaranteed to be preserved.
```

Comment:
```suggestion
* -   The algorithm is not stable, meaning that the relative order of non-NaN elements is not guaranteed to be preserved.
```

If we are using this for `dquicksort`, the "instability" doesn't cause an issue. We can also add something like `dsortnans2` (if needed) later which would preserve the relative order of non-NaN elements.